e2e/condition_service_test.go

```diff
@@ -15,99 +15,130 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
-// TestCondition_Services_Regexp runs the CTS binary to test
-// a task configured with a regexp in the service condition.
+// TestCondition_Services runs the CTS binary to test a task
+// configured with regexp/names in the service condition.
 // This test confirms that when a service is registered that
-// doesn't match the task condition's regexp config, no task
+// doesn't match the task condition's regexp/names config, no task
 // is triggered.
-func TestCondition_Services_Regexp(t *testing.T) {
+func TestCondition_Services(t *testing.T) {
 	t.Parallel()
 
-	srv := testutils.NewTestConsulServer(t, testutils.TestConsulServerConfig{
-		HTTPSRelPath: "../testutils",
-	})
-	defer srv.Stop()
-
-	tempDir := fmt.Sprintf("%s%s", tempDirPrefix, "services_condition_regexp")
-	taskName := "services_condition_task"
-	conditionTask := fmt.Sprintf(`task {
-	name = "%s"
-	source = "./test_modules/local_instances_file"
-	condition "services" {
-		regexp = "api-"
-		filter = "Service.Tags not contains \"tag_a\""
-		cts_user_defined_meta {
-			my_meta_key = "my_meta_value"
-		}
+	cases := []struct {
+		name       string
+		taskName   string
+		taskConfig string
+	}{
+		{
+			"services condition - regexp",
+			"services_condition_regexp",
+			`task {
+				name = "%s"
+				source = "./test_modules/local_instances_file"
+				condition "services" {
+					regexp = "api-"
+					filter = "Service.Tags not contains \"tag_a\""
+					cts_user_defined_meta {
+						my_meta_key = "my_meta_value"
+					}
+				}
+			}`,
+		},
+		{
+			"services condition - names",
+			"services_condition_names",
+			`task {
+				name = "%s"
+				source = "./test_modules/local_instances_file"
+				condition "services" {
+					names  = ["api-web"]
+					filter = "Service.Tags not contains \"tag_a\""
+					cts_user_defined_meta {
+						my_meta_key = "my_meta_value"
+					}
+				}
+			}`,
+		},
+	}
+
+	for _, tc := range cases {
+		tc := tc // rebind tc into this lexical scope for parallel use
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel() // Run table tests in parallel as they can take a lot of time
```

Comment:
Nice scoping! (no-op for the PR but to keep in mind) This flags to me we may have this bite us elsewhere in our tests 🤔

Reply:
Can't take credit for this! I borrowed this pattern introduced earlier.

Yea, the issue that I know of is if someone tries to runs e2e test containing t.Parallel (or even some a subset of e2e tests) locally, it will have a race issue when the CTS starts in parallel (I think around downloading TF?!). Last I know for running all tasks (as in circleci), we have one test that runs first to avoid this race. You might have something else in mind though.

Reply:
I see. Nah I haven't sifted through the code to have a specific case to update, but it could be something I wasn't keen on myself in one of my tests. I'll keep my eye out if I come across this pattern for loop + parallel.